daemon/cluster/helpers.go

```diff
@@ -95,14 +101,45 @@ func getTask(ctx context.Context, c swarmapi.ControlClient, input string) (*swar
 		}
 
 		if len(rl.Tasks) == 0 {
-			return nil, fmt.Errorf("task %s not found", input)
+			err := fmt.Errorf("task %s not found", input)
+			return nil, errors.NewRequestNotFoundError(err)
 		}
 
 		if l := len(rl.Tasks); l > 1 {
-			return nil, fmt.Errorf("task %s is ambiguous (%d matches found)", input, l)
+			err := fmt.Errorf("task %s is ambiguous (%d matches found)", input, l)
+			return nil, errors.NewRequestConflictError(err)
 		}
 
 		return rl.Tasks[0], nil
 	}
 	return rg.Task, nil
 }
+
+func getNetwork(ctx context.Context, c swarmapi.ControlClient, input string) (*swarmapi.Network, error) {
+	// GetNetwork to match via full ID.
+	rg, err := c.GetNetwork(ctx, &swarmapi.GetNetworkRequest{NetworkID: input})
+	if err != nil {
+		// If any error (including NotFound), ListNetworks to match via ID prefix and full name.
+		rl, err := c.ListNetworks(ctx, &swarmapi.ListNetworksRequest{Filters: &swarmapi.ListNetworksRequest_Filters{Names: []string{input}}})
+		if err != nil || len(rl.Networks) == 0 {
+			rl, err = c.ListNetworks(ctx, &swarmapi.ListNetworksRequest{Filters: &swarmapi.ListNetworksRequest_Filters{IDPrefixes: []string{input}}})
+		}
+
+		if err != nil {
+			return nil, err
+		}
+
+		if len(rl.Networks) == 0 {
+			err := fmt.Errorf("network %s not found", input)
+			return nil, errors.NewRequestNotFoundError(err)
+		}
+
+		if l := len(rl.Networks); l > 1 {
+			err := fmt.Errorf("network %s is ambiguous (%d matches found)", input, l)
+			return nil, errors.NewRequestConflictError(err)
```

Comment:
Conflict errors make more sense when you try to modify something and something else has since changed it.

A 300 might be a better fit.